tests/js/client/shell/shell-debug-dump.js

```diff
@@ -0,0 +1,259 @@
+/* jshint globalstrict:false, strict:false, maxlen: 200 */
+/* global assertEqual, assertTrue, assertNull, assertNotNull, fail */
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2010-2023 triagens GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License")
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+/// @author Copyright 2023, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require('jsunity');
+const internal = require('internal');
+const fs = require('fs');
+const arangodb = require('@arangodb');
+const explainer = require("@arangodb/aql/explainer");
+const db = arangodb.db;
+const isEnterprise = require("internal").isEnterprise();
+let smartGraph = null;
+if (isEnterprise) {
+  smartGraph = require("@arangodb/smart-graph");
+}
+const graphModule = require("@arangodb/general-graph");
+
+const vn1 = "testVertex1";
+const vn2 = "testVertex2";
+const vn3 = "testVertex3";
+const cn1 = "edgeTestCollection";
+const cn2 = "edgeTestCollection2";
+const relationName = "isRelated";
+const graphName = "myGraph";
+const dbName = "testDatabase";
+const fileName = fs.getTempFile() + "-debugDump";
+const outFileName = fs.getTempFile() + "-inspectDump";
+
+
+function recreateEmptyDatabase () {
+  db._useDatabase("_system");
+  db._dropDatabase(dbName);
+  db._createDatabase(dbName);
+  db._useDatabase(dbName);
+  try {
+    db._graphs.document(graphName);
+    fail();
+  } catch (err) {
+    assertEqual(err.errorMessage, "document not found");
+  }
+}
+
+function createOrderedKeysArray (collName) {
+  let keys = db[collName].toArray().map(doc => doc._key);
+  keys.sort();
+  return keys;
+}
+
+function testGeneralWithSmartGraphNamed (amountExamples) {
+  const graphBefore = db._graphs.document(graphName);
+  delete graphBefore._rev;
+  const query = `FOR v,e, p IN 0..10 ANY 'customer/customer-abc' GRAPH '${graphName}'  RETURN {key: e._key}`;
+  explainer.debugDump(fileName, query, {}, {examples: amountExamples});
+  explainer.inspectDump(fileName, outFileName);
+  recreateEmptyDatabase();
+  internal.load(outFileName);
+  assertNull(db[vn1].any());
```

Comment:
This is only correct if we are called without examples right?
Should we have a test with above query and examples?

Reply:
Will add an example

Reply:
Oh no, this one has an example already